src/js/views/globals/search/patient-search_views.js

```diff
@@ -42,11 +42,34 @@ const PatientSearchPicklist = Picklist.extend({
   itemTemplate: hbs`
     {{matchText text search}}{{~ remove_whitespace ~}}
     <span class="patient-search__picklist-item-meta">{{formatDateTime birth_date "MM/DD/YYYY"}}</span>
+    {{#each identifiers}}<span class="patient-search__picklist-item-meta">{{this}}</span>{{/each}}
   `,
   itemTemplateContext() {
+    // REMOVE: for sandbox testing purposes only
+    // this.model.set(
+    //  'identifiers',
+    //  [{ 'type': 'PID', 'value': '0123456789' }, { 'type': 'SSN', 'value': '321-45-6789' }, { 'type': 'XID', 'value': '9876543210' }],
+    // );
+
+    const settings = this.state.get('settings');
+    const settingsIdentifiers = settings ? settings.result_identifiers : [];
+
+    const patientIdentifiers = this.model.get('identifiers');
+
+    const identifiers = [];
+
+    each(settingsIdentifiers, settingsIdentifier => {
+      const identifierToShow = find(patientIdentifiers, patientIdentifier => {
+        return settingsIdentifier.toUpperCase() === patientIdentifier.type.toUpperCase();
+      });
+
+      if (identifierToShow) identifiers.push(identifierToShow.value);
+    });
+
     return {
       text: this.getItemSearchText(),
       search: this.state.get('search'),
+      identifiers,
```

Comment:
Breakdown of what I did here:

1. Get the list of identifiers set in the `patient_search_settings` organization setting. And store it in the `settingsIdentifiers` variable. 
2. Get the patient's `identifiers` data and store it in the `patientIdentifiers` variable. My assumption is that that data will be structured like this: `[{ 'type': 'PID', 'value': '0123456789' }, { 'type': 'SSN', 'value': '321-45-6789' }]`.
3. Loop through each identifier specified in the org setting and find it's match in the patient's identifier data. Only push it to the `identifiers` array if a match is found. I'm using `toUpperCase()` because the org setting might contain lowercase versions of the identifier type (e.g. `["mrn", "ssn"]`)?
4. Update the handlebars template to display each identifier value in the `identifiers` array. If the `identifiers` array is empty, no identifiers will be displayed.

I tried to move this code into its own function (e.g. `this.getIdentifiers()`) and do this instead:

```js
return {
  text: this.getItemSearchText(),
  search: this.state.get('search'),
  identifiers: this.getIdentifiers(),
}
```

But I was getting `this.getIdentifiers() is not defined` errors and couldn't figure out why... 😕 

So not entirely sure where to put this code.

Reply:
Yeah the `item___` function context is odd, but the `this` is the item view of the list and not the parent component.